package.json

```diff
@@ -4,7 +4,13 @@
   "description": "JavaScript implementation of the IPLD raw format.",
   "leadMaintainer": "Volker Mische <volker.mische@gmail.com>",
   "main": "src/index.js",
+  "types": "./dist/src/index.d.ts",
+  "files": [
```

Comment:
I don't think this is necessary for types, can we not do this?

Reply:
`./dist` is in `.gitignore` so npm ignores it when publishing.

You either have to add an `.npmignore` file that has everthing `.gitignore` has except `./dist`, or you have to add a `files` array to the `package.json`.

With `files`:

```console
$ npm pack
npm notice 
npm notice 📦  ipld-raw@6.0.0
npm notice === Tarball Contents === 
npm notice 1.1kB  LICENSE                
npm notice 2.4kB  src/index.js           
npm notice 58.8kB dist/index.min.js         <--- bundle included
npm notice 1.6kB  package.json           
npm notice 107B   dist/src/index.d.ts.map   <--- types included
npm notice 4.4kB  CHANGELOG.md           
npm notice 2.8kB  README.md              
npm notice 3.0kB  dist/src/index.d.ts    
npm notice === Tarball Details === 
npm notice name:          ipld-raw                                
npm notice version:       6.0.0                                   
npm notice filename:      ipld-raw-6.0.0.tgz                      
npm notice package size:  23.8 kB                                 
npm notice unpacked size: 74.2 kB                                 
npm notice shasum:        317efc556fa2469c21e2667f0a2a4b4f894de968
npm notice integrity:     sha512-2KswCk1D0beqZ[...]VhSAtmDTc7XDA==
npm notice total files:   8                                       
npm notice 
```

Without `files`:

```console
$ npm pack
npm notice 
npm notice 📦  ipld-raw@6.0.0
npm notice === Tarball Contents === 
npm notice 1.1kB  LICENSE           
npm notice 2.4kB  src/index.js      
npm notice 2.3kB  test/index.spec.js
npm notice 1.5kB  package.json      
npm notice 162B   tsconfig.json     
npm notice 4.4kB  CHANGELOG.md      
npm notice 2.8kB  README.md         
npm notice 23.8kB ipld-raw-6.0.0.tgz
npm notice 685B   .travis.yml       
npm notice === Tarball Details === 
npm notice name:          ipld-raw                                
npm notice version:       6.0.0                                   
npm notice filename:      ipld-raw-6.0.0.tgz                      
npm notice package size:  29.8 kB                                 
npm notice unpacked size: 39.2 kB                                 
npm notice shasum:        2038ce0d9b5efb35de706569066ac8f48ca042ea
npm notice integrity:     sha512-Yo5KGfhVUbvDp[...]U/yUg/ZuZ/QdA==
npm notice total files:   9                                       
npm notice 
```

Reply:
🤦 
for the record, I think tests and friends should all just be bundled as a matter of consistency, but I don't care strongly enough to hold this up

Reply:
I sort of agree, as long as everyone is well behaved - if your tests contain lots of fixtures it can make the installs very heavy and cause useful tools to [stop working](https://github.com/EqualMa/gitpkg/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed++502)